assignments/week_1/exercise_1_1/exercise_1_1_answer.cpp

```diff
@@ -0,0 +1,62 @@
+//Do not edit the next 5 lines
+#include <stdlib.h>
+#include <iostream>
+#include <vector>
+
+int main(int argc, char** argv) {
+    std::cout << "Use the wasd keys and enter to input a direction, or the E key to stop" << std::endl;
+
+    // Write your code for step 1 here
+    std::vector<float> wheel_speeds(4);
+    // Write your code for step 2 here
+    wheel_speeds[0] = 0;
+    wheel_speeds[1] = 0;
+    wheel_speeds[2] = 0;
+    wheel_speeds[3] = 0;
+    // Do not edit the next 5 lines
+    bool keep_going = true;
+    char direction;
+
+    while (keep_going) {
+        std::cin >> direction; //At the begining of every iteration look for an input from the user
+        // Write rest of code here (within while loop)
+        if (direction == 'w') {
+            for (int i = 0; i < 4; i++) {
```

Comment:
Wouldn't that mean having to use the & operator since the range based loop is pass by value? I can include it if we want, but new members might get confused. I could use it for the last one where they just print out the wheel speeds but then they don't get the 'number' of the wheel so easily.

Reply:
Yeah, in this particular example it would. @barulicm this is also the solution, so we changing things here doesn't really give them practice with that.

Reply:
The loop below that prints out the wheel speeds could be done without references, so that would be a good one to choose. You can encourage them to use the range-based loop style in the instructions.

Reply:
The thing is it then it becomes difficult to print out the index of the wheel with the wheel speed in a nice way, which I think is probably a useful way for them to see if their code works.

Reply:
You could just as easily ask them to print it out as a vector, like `{w0, w1, w2, w3}`. This would allow them to use range-based for.

Reply:
@HussainGynai that's a good point about printing out the index. I missed that it was being used that way. Alright, maybe just forget about that for this exercise. I'll keep an eye out for opportunities to put them in other exercises.